### ISSUE TITLE
Warn about sed pattern variable expansion with unescaped forward shashes

### ISSUE THREAD

**Issue by karolinepauls** [koalaman/shellcheck#1623]
#### For new checks and feature suggestions
- [ v ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ v ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
AAA='AAA/AAA'
sed "s/12/$AAA/"
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

```
Line 3:
sed "s/12/$AAA/"
^-- SCXXXX: Variable expansion in sed patterns should escape forward slashes, like: "${VARIABLE//\//\\/}"; ignore if variable is already escaped.
```

Not all shells can use character substitution when expanding variables.

It may be hard to detect if forward slashes are escaped correctly but it's easy to detect they're not escaped at all.

False positives: without the "ignore if variable is already escaped" clause, if the string is already escaped, we risk suggesting to double-escape it. However, if it contains no slashes, escaping them won't change anything.

There may be better ways to escape forward slashes.

**Comment by ISSOtm** [comment]
I am hitting an expansion of this problem, where this is also wrong:

```sh
#!/bin/bash
file=b.sh
sed "s/$file/<placeholder>/" # Will actually replace any instance of e.g. `bash`, not just `b.sh`!
```

...unfortunately I recognise that this may be difficult to implement. Nonetheless, I think it's a good idea to have this warning (class?), since “the variable was escaped previously, ignore this” is the kind of relatively uncommon thing that `#shellcheck disable` is perfect for :)

I would thus suggest your warning to be worded like “parameter expansion in sed substitutions should escape delimiters”. The warning that I am suggesting is a more general “parameter expansion should escape regex metacharacters” (...but I fear that the list of these metacharacters is going to be a rabbit hole and a half, lol.)

Oh, on that note: `sed` does *not* require slashes for `s`! Any character is accepted, so your example can even be written `sed s,12,$AAA,` and then the required escaping would be `${AAA//,/\\,}`.

(I would love to implement those warnings myself, but I don't know Haskell and fear that they are complex enough that I wouldn't be able to crib other checks 😞)
